Sources/PackageSigning/VerifierPolicies.swift

```diff
@@ -0,0 +1,106 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift open source project
+//
+// Copyright (c) 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import struct Foundation.Data
+import struct Foundation.Date
+import struct Foundation.URL
+
+import Basics
+@_implementationOnly import X509
+
+extension SignatureProviderProtocol {
+    func buildPolicySet(configuration: VerifierConfiguration, httpClient: HTTPClient) -> PolicySet {
+        var policies = [VerifierPolicy]()
+
+        if case .enabled(let validationTime) = configuration.certificateExpiration {
+            policies.append(RFC5280Policy(validationTime: validationTime ?? Date()))
+        }
+
+        switch configuration.certificateRevocation {
+        case .strict:
+            policies.append(_OCSPVerifierPolicy(httpClient: httpClient, mode: .strict))
+        case .allowSoftFail:
+            policies.append(_OCSPVerifierPolicy(httpClient: httpClient, mode: .allowSoftFail))
+        case .disabled:
+            ()
+        }
+
+        return PolicySet(policies: policies)
+    }
+}
+
+struct _OCSPVerifierPolicy: VerifierPolicy {
+    private let underlying: OCSPVerifierPolicy<_OCSPRequester>
+    private let mode: Mode
+
+    init(httpClient: HTTPClient, mode: Mode) {
+        self.underlying = OCSPVerifierPolicy(requester: _OCSPRequester(httpClient: httpClient))
+        self.mode = mode
+    }
+
+    func chainMeetsPolicyRequirements(chain: UnverifiedCertificateChain) async -> PolicyEvaluationResult {
+        let result = await self.underlying.chainMeetsPolicyRequirements(chain: chain)
+        switch result {
+        case .meetsPolicy:
+            return result
+        case .failsToMeetPolicy(let reason):
+            switch self.mode {
+            case .strict:
+                return result
+            case .allowSoftFail:
+                // Allow 'unknown' status and failed OCSP request in this mode
+                guard !reason.lowercased().contains("revoked through ocsp") else {
+                    return result
```

Comment:
@Lukasa @dnadoba Can you think of a better way to allow soft fail?

Reply:
Hmm, yeah, good question. I think what we ideally want here is actually configuration on the OCSP policy itself. @dnadoba want to add that support?